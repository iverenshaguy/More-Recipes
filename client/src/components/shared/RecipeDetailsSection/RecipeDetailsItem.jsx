import React from 'react';
import PropTypes from 'prop-types';

/**
 * @exports
 * Returns Recipe Details Item
 * @param {object} props
 * @returns {JSX} RecipeDetailsItem
 */
const RecipeDetailsItem = ({ section }) => (
  <div className="p-xs-1 px-md-5 pt-3">
    <div className="row">
      <div className="col-xs-12 col-md-4 col-lg-4">
        <h4 className="uppercase">{section.title}</h4>
      </div>
      <div className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1" id="recipe-details-info">
        {section.info}
      </div>
    </div>
    <hr />
    <div className="text-justify" id={section.id}>
      {section.items.length === 0 &&
        <p className="mb-3">There are no {section.title.toLowerCase()}.</p>}
      {section.items &&
        section.items.map(item => <p className="mb-3" key={item}> {item}</p>)}
    </div>
  </div>
);

RecipeDetailsItem.propTypes = {
  section: PropTypes.shape({
    items: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    info: PropTypes.element
  }).isRequired
};

export default RecipeDetailsItem;
