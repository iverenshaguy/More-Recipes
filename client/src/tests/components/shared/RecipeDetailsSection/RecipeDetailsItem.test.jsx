import React from 'react';
import RecipeDetailsItem from '../../../../components/shared/RecipeDetailsSection/RecipeDetailsItem';

const section = {
  items: ['good', 'nice'],
  title: 'Cooking Directions',
  id: 'cooking-directions',
  info: <p>I love to eat</p>
};

describe('RecipeDetailsSection', () => {
  it('renders correctly', () => {
    const wrapper = mount(<RecipeDetailsItem section={section} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders message when item is empty', () => {
    const wrapper = mount(<RecipeDetailsItem section={{ ...section, items: [] }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p.mb-3').text()).toEqual('There are no cooking directions.');
    wrapper.unmount();
  });
});
