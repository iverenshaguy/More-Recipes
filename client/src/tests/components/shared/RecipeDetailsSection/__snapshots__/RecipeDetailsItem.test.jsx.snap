// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsSection renders correctly 1`] = `
<RecipeDetailsItem
  section={
    Object {
      "id": "cooking-directions",
      "info": <p>
        I love to eat
      </p>,
      "items": Array [
        "good",
        "nice",
      ],
      "title": "Cooking Directions",
    }
  }
>
  <div
    className="p-xs-1 px-md-5 pt-3"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-4 col-lg-4"
      >
        <h4
          className="uppercase"
        >
          Cooking Directions
        </h4>
      </div>
      <div
        className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1"
        id="recipe-details-info"
      >
        <p>
          I love to eat
        </p>
      </div>
    </div>
    <hr />
    <div
      className="text-justify"
      id="cooking-directions"
    >
      <p
        className="mb-3"
        key="good"
      >
         
        good
      </p>
      <p
        className="mb-3"
        key="nice"
      >
         
        nice
      </p>
    </div>
  </div>
</RecipeDetailsItem>
`;

exports[`RecipeDetailsSection renders message when item is empty 1`] = `
<RecipeDetailsItem
  section={
    Object {
      "id": "cooking-directions",
      "info": <p>
        I love to eat
      </p>,
      "items": Array [],
      "title": "Cooking Directions",
    }
  }
>
  <div
    className="p-xs-1 px-md-5 pt-3"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-4 col-lg-4"
      >
        <h4
          className="uppercase"
        >
          Cooking Directions
        </h4>
      </div>
      <div
        className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1"
        id="recipe-details-info"
      >
        <p>
          I love to eat
        </p>
      </div>
    </div>
    <hr />
    <div
      className="text-justify"
      id="cooking-directions"
    >
      <p
        className="mb-3"
      >
        There are no 
        cooking directions
        .
      </p>
    </div>
  </div>
</RecipeDetailsItem>
`;
